tests/lib/smf/src/test_lib_flat_smf.c

```diff
@@ -223,15 +233,19 @@ void test_smf_flat(void)
 {
 	/* A) Test transitions */
 
+	int ret;
 	test_obj.transition_bits = 0;
 	test_obj.terminate = NONE;
 	smf_set_initial((struct smf_ctx *)&test_obj, &test_states[STATE_A]);
 
-	for (int i = 0; i < SMF_RUN; i++) {
-		if (smf_run_state((struct smf_ctx *)&test_obj)) {
-			break;
-		}
-	}
+	ret = smf_run_state((struct smf_ctx *)&test_obj);
+	zassert_equal(0, ret, "Incorrect smf_run_state return value");
+	ret = smf_run_state((struct smf_ctx *)&test_obj);
+	zassert_equal(0, ret, "Incorrect smf_run_state return value");
+	ret = smf_run_state((struct smf_ctx *)&test_obj);
+	zassert_equal(0, ret, "Incorrect smf_run_state return value");
+	ret = smf_run_state((struct smf_ctx *)&test_obj);
+	zassert_equal(1, ret, "Incorrect smf_run_state return value");
```

Comment:
Because the tests below check that the final state gets reach, I prefer calling `smf_run_state()` in a loop, as this better matches what application code will do.

Reply:
fixed using for loop.  Transition tests run SMF_NUM_STATES times, while each exit code test will run 1, 2, or 3 times, depending on the exit state (ENTRY, RUN, EXIT respectively)